common/src/ux/codec.rs

```diff
@@ -1,20 +1,23 @@
 use alloc::{string::String, vec::Vec};
-use core::convert::TryInto;
+use core::{convert::TryInto, mem::MaybeUninit};
 
 pub trait Serializable {
     fn get_serialized_length(&self) -> usize;
-    fn serialize(&self, buf: &mut [u8], pos: &mut usize);
+    fn serialize(&self, buf: &mut [MaybeUninit<u8>], pos: &mut usize);
```

Comment:
Why does this trait take `pos` rather than just `buf` (possibly as `&mut &mut [MaybeUninit<u8>]`) and let the caller adjust the position only if needed? I think it should remove a bunch of checks.